File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-contract/src/test/java/de/invesdwin/scripting/haskell/runtime/contract/callback/ParametersAndReturnsTestCharacter.hs
println("getCharacter")
if isdefined(Main, :getCharacter) && !isnothing(getCharacter)
	error("getCharacter already defined!")
end
getCharacter = callback("getCharacter")
println(typeof(getCharacter))
println(getCharacter)
if typeof(getCharacter) != Char
	error("getCharacter not Char!")
end
callback("setCharacter",getCharacter)

println("getCharacterVector")
if isdefined(Main, :getCharacterVector) && !isnothing(getCharacterVector)
	error("getCharacterVector already defined!")
end
getCharacterVector = callback("getCharacterVector")
println(eltype(getCharacterVector))
println(getCharacterVector)
if eltype(getCharacterVector) != Char
	error("getCharacterVector not Char!")
end
callback("setCharacterVector",getCharacterVector)

println("getCharacterVectorAsList")
if isdefined(Main, :getCharacterVectorAsList) && !isnothing(getCharacterVectorAsList)
	error("getCharacterVectorAsList already defined!")
end
getCharacterVectorAsList = callback("getCharacterVectorAsList")
println(eltype(getCharacterVectorAsList))
println(getCharacterVectorAsList)
if eltype(getCharacterVectorAsList) != Char
	error("getCharacterVectorAsList not Char!")
end
callback("setCharacterVectorAsList",getCharacterVectorAsList)

println("getCharacterMatrix")
if isdefined(Main, :getCharacterMatrix) && !isnothing(getCharacterMatrix)
	error("getCharacterMatrix already defined!")
end
getCharacterMatrix = callback("getCharacterMatrix")
println(eltype(getCharacterMatrix))
println(getCharacterMatrix)
if eltype(getCharacterMatrix) != Char
	error("getCharacterMatrix not Char!")
end
callback("setCharacterMatrix",getCharacterMatrix)

println("getCharacterMatrixAsList")
if isdefined(Main, :getCharacterMatrixAsList) && !isnothing(getCharacterMatrixAsList)
	error("getCharacterMatrixAsList already defined!")
end
getCharacterMatrixAsList = callback("getCharacterMatrixAsList")
println(eltype(getCharacterMatrixAsList))
println(getCharacterMatrixAsList)
if eltype(getCharacterMatrixAsList) != Char
	error("getCharacterMatrixAsList not Char!")
end
callback("setCharacterMatrixAsList",getCharacterMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-rust/invesdwin-scripting-rust-runtime-contract/src/test/java/de/invesdwin/scripting/rust/runtime/contract/callback/ParametersAndReturnsTestCharacter.rs
println!("getCharacter");
let getCharacter: char = callback("getCharacter", &[]);
println!("getCharacter: {}", getCharacter);
callback_void("setCharacter", &[param(getCharacter)]);

println!("getCharacterVector");
let getCharacterVector: Vec<char> = callback("getCharacterVector", &[]);
println!("getCharacterVector: {:?}", getCharacterVector);
callback_void("setCharacterVector", &[param(getCharacterVector)]);

println!("getCharacterVectorAsList");
let getCharacterVectorAsList: Vec<char> = callback("getCharacterVectorAsList", &[]);
println!("getCharacterVectorAsList: {:?}", getCharacterVectorAsList);
callback_void("setCharacterVectorAsList", &[param(getCharacterVectorAsList)]);

println!("getCharacterMatrix");
let getCharacterMatrix: Vec<Vec<char>> = callback("getCharacterMatrix", &[]);
println!("getCharacterMatrix: {:?}", getCharacterMatrix);
callback_void("setCharacterMatrix", &[param(getCharacterMatrix)]);

println!("getCharacterMatrixAsList");
let getCharacterMatrixAsList: Vec<Vec<char>> = callback("getCharacterMatrixAsList", &[]);
println!("getCharacterMatrixAsList: {:?}", getCharacterMatrixAsList);
callback_void("setCharacterMatrixAsList", &[param(getCharacterMatrixAsList)]);
